### ISSUE TITLE
Type completion for variable instantation

### ISSUE THREAD

**Issue by aaronkavlie-wf** [redhat-developer/vscode-java#118]
Is there a way to autocomplete the type when creating a variable?
If not, there should be!

```java
// 1. Create a var without specifying type (let's say someMethod returns type int)
foo = someObject.someMethod();
// 2. [some magic shortcut]
// 3. Replaced with:
int foo = someObject.someMethod();
```

[Discussion about IntelliJ functionality for this](http://stackoverflow.com/questions/2070653/what-is-the-intellij-shortcut-to-create-a-local-variable)

**Comment by gorkem** [comment]
This does not exist as code completion and I always felt that it should. We do have plans for adding quick fixes https://github.com/gorkem/java-language-server/issues/15, I think we can have this in as part of that work.

**Comment by aaronkavlie-wf** [comment]
That would be awesome @gorkem.

I tried out IntelliJ just for this feature, and sure enough it handles this (via a refactoring, which is more complicated than it ought to be), but I'd rather use VS Code if I can.

**Comment by fbricon** [comment]
This was implemented in 0.7.0. Use `Ctrl+'.'` (`Cmd+'.' on MacOS`) to enable code actions :

![create-missing-var](https://user-images.githubusercontent.com/148698/32816461-c2ef2a1c-c986-11e7-8972-84976f37a35a.gif)
